app/controllers/certificates_controller.rb

```diff
@@ -0,0 +1,81 @@
+class CertificatesController < ApplicationController
+  before_action :authorise_manage_certificates
+  before_action :authorise_manage_current_certificate, only: %i[show edit destroy]
+
+  def index
+    @pagy, @certificates = pagy(Certificate.where(organisation: current_organisation))
+  end
+
+  def show
+    @certificate = Certificate.find(params[:id])
+  end
+
+  def edit
+    @certificate = Certificate.find(params[:id])
+  end
+
+  def destroy
+    @certificate = Certificate.find(params[:id])
+    cert_name = @certificate.name
+    @certificate.destroy!
+    Services.certificate_repository.delete_certificate(current_organisation.id, cert_name, @certificate.is_root_cert)
+    redirect_to(certificates_path, notice: "Successfully removed Certificate: #{@certificate.name}")
+  end
+
+  def new
+    @certificate = Certificate.new(organisation: current_organisation)
+  end
+
+  def create
+    uploaded_file = params.dig(:certificate, :cert)
+    cert_name = params.dig(:certificate, :name)
+
+    if uploaded_file
+      flash.clear
+      raw_cert = uploaded_file.read
+      mime_type = Marcel::MimeType.for(uploaded_file)
+
+      if mime_type.to_s == "application/x-x509-ca-cert"
+        @certificate = Certificate.new(organisation: current_organisation, name: cert_name)
+        begin
+          @certificate.import_from_x509_content(raw_cert)
+        rescue RuntimeError => e
```

Comment:
Maybe better to catch all errors at the end of this method?

Reply:
done